app/src/main/java/chat/rocket/android/activity/ServerConfigActivity.java

```diff
@@ -77,14 +83,27 @@ protected void onPause() {
   }
 
   private void onRenderServerConfigError(ServerConfig config) {
-    if (config != null) {
+    if (config.getState() == ServerConfig.STATE_CONNECTION_ERROR) {
```

Comment:
Why the logic of 'state = STATE_CONNECTION_ERROR' is moved?
By moving the logic, sessionObserver.sub() is called repeatedly.